lib/spack/spack/solver/concretize.lp

```diff
@@ -855,156 +870,168 @@ opt_criterion(100, "number of packages to build (vs. reuse)").
 #defined optimize_for_reuse/0.
 
 % Minimize the number of deprecated versions being used
-opt_criterion(15, "deprecated versions used").
-#minimize{ 0@215: #true }.
-#minimize{ 0@15: #true }.
+opt_criterion(75, "deprecated versions used").
+#minimize{ 0@275: #true }.
+#minimize{ 0@75: #true }.
 #minimize{
-    1@15+Priority,Package
+    1@75+Priority,Package
     : deprecated(Package, _),
       build_priority(Package, Priority)
 }.
 
+% A condition group specifies one or more specs that must be satisfied.
+% Specs declared first are preferred, so we assign increasing weights and
+% minimize the weights.
+opt_criterion(73, "requirement weight").
+#minimize{ 0@273: #true }.
+#minimize{ 0@73: #true }.
+#minimize {
+    Weight@73+Priority
+    : condition_weight(Package, Weight),
+      build_priority(Package, Priority)
+}.
```

Comment:
@becker33 we spoke about giving this the lowest priority, but if this isn't prioritized above e.g. version weights, then for example if someone has

```
require:
  one_of: [1.0, 1.1]
```

Then Spack will pick `1.1` - the user can't control this preference via `requires`. I think they could combine this with `version: [1.0, 1.1]` but IMO they should be able to control it with just the `requires` section.

That being said, if the ramifications of placing this last are understood and still desired, then I can definitely do that.

I also added some "space" between the priorities, to make it easier to insert new ones in the future.